### DISCUSSION TITLE
Correcting pre-tracked cell lineage

### DISCUSSION THREAD

**Discussion by mm809** [SchmollerLab/Cell_ACDC#205] [Q&A]
Hi Francesco,

Thank you for adding the new utilities! I am hoping to correct pre-tracked cell lineages alongside tracking. But it seems that the lineage correction in ACDC is done in the 'Cell Cycle analysis' mode in ACDC and the pre-tracked lineage is somehow not shown up there. Probably I'm missing something obvious but please let me know...

We are mostly working with mammalian cells that divide symmetrically. The lineage relationship is a bit different from yeast, where mother-bud relationships capture almost all lineage information (a mixture of sister and parent-daughter relationships in mammalian systems). The parent-daughter relationship is the most critical one that can define the lineage tree in our cells. The parent cell and the two daughters have their own ID. At the frame of cell division, the parent ID track ends and the two daughter tracks begin. We often see the following errors in cell lineage relationships -
1. cell division is not recognized. The parent cell ID is lost and two new cells appear.
2. cell division is not recognized. One daughter cell keeps the parent's ID and one gets a new cell ID.
3. cell division is correctly recognized, but a nearby cell is incorrectly assigned as the daughter cell.
4. No cell division occurs but one is incorrectly assigned.

It would be great to know if ACDC can be used to do these corrections.

Thank you very much.
Mingwei

**Comment by ElpadoCan** [comment]
Hi @mm809, 

The way you would do cell cycle annotation for symmetrically dividing cells in Cell-ACDC is not super-intuitive at first, but it's doable. We have plans to improve that, but it will take some time.

However, it is possible, we have users doing that. They are not doing it from pre-tracked lineage. I don't see why it should not work with a little bit of fine-tuning.

So the way it works for budding yeast is actually more complicated than symmetrically dividing cells because in budding yeast we need annotate the start of the budding phase and end of the budding phase, so there are two events per cycle that need to be annotated. In symmetrically dividing cells instead, there is only one event, which is division. 

The problem is that, to show the parent-daughter relationship (let's call it PDr to write faster) Cell-ACDC wants to draw a line connecting P with D. However, this is not possible of course because the parent is in the previous frame. The way we solved this is by keeping the same ID of P to one of the two daughters. 

This way, when you want to correct the PDr, you click on the daughter and click again (or drag and drop) on the right parent. Note that this doesn't mean that division is not recognized, because you then right-click on either the parent or the daughter to annotate division on that frame. You will then see the line connecting P with D disappearing and Cell-ACDC will annotate the division frame and it will increase the generation number accordingly. 

This is also required because the wrong parent needs to go back one cycle, so the generation number of the wrong parent will also be automatically corrected.

Now, to start from pre-computed relationships I need to fine-tune the process a little bit which means the following:

1. The segmentation mask should have IDs where one of the two daughters keeps the same ID as the parent ID. Basically, once an ID appears will stay for the entire rest of the video (I call this the root_ID).

2. We need to keep at least one frame where the cells are related to each other but not divided yet, otherwise Cell-ACDC will not draw the line connecting them. This means that the division will be annotated one frame after. It is possible to annotate on the same frame, but that would be the standard way of annotating PDRs from scratch and not from pre-computed relationships.

This process gives you the possibility to annotate more than just PDRs and division. You can also use it to annotate the start of other phases because you can have some frames in which two cells are related but not divided yet (as long as they are segmented separately).

Once you are happy with annotations we have a utility that adds additional columns specific for symmetrically dividing cells. These columns all end with `_tree` in the name referring to the lineage tree. These columns are:

1. `Cell_ID_tree`: a new ID for each daughter cell upon division (mother ID doesn't exist anymore).
2. `parent_ID_tree`: the ID of the parent
3. `sister_ID_tree`: the ID of the sister cell
4. `root_ID_tree`: the ID of the very first cell that started the entire tree
5. `generation_num_tree`: the generation number in the tree, basically the number of divisions

These columns are what one would expect when annotating lineage in symmetrically dividing cells, but they are not easily visualisable and correctable in a GUI. Hence, the strategy I outlined above. With these columns, you can then do downstream analysis and extract information like cell cycle duration, and the number of divisions or plot numerical features over the cell cycle, like cell volume, cell growth, growth rate and features from fluorescent channels. We have a Jupyter notebook written in Python to show how to do those calculations. 

Now let me answer your points:

> 1. cell division is not recognized. The parent cell ID is lost and two new cells appear.

This is a segmentation error, so you have to first correct this in Segmentation and tracking mode. 

> 2. cell division is not recognized. One daughter cell keeps the parent's ID and one gets a new cell ID.

Given the way Cell-ACDC deals with visualization of the PDRs, this is fine with segmentation and you would simply correct the relationship and then right-click to actually annotate the division.

> 3. cell division is correctly recognized, but a nearby cell is incorrectly assigned as the daughter cell.

Here you would click on the daughter you want to correct and release on the correct daughter cell

> 4. No cell division occurs but one is incorrectly assigned.

Here you would right-click on the wrong assignment to undo it.

So, going forward, you cannot do what you are planning to do right now. We first need to either make a new utility or change the behaviour of the one I just made. This utility should create a single frame where two cells are related to each other but not divided yet, and it should create segmentation masks where one of the two daughters keeps the parent ID. You would then create the right IDs (the `tree` columns) with the utility I described above. 

I hope it's clear enough and let me know what you think about it. If you think it might be a suitable strategy for you, I can implement it.

I know it sounds a bit complicated, but at the moment I cannot think of a better way unless we write a completely new module for the cell cycle analysis of symmetrical dividing cells. We have plans to do that, and hopefully, we can do it soon if funding applications go well. 

Cheers,
Francesco

**Reply by mm809** [reply]
Hi Francesco,

Thanks for explaining the linking logic. It sounds like if I change the tabular input so that one daughter keeps the parent ID and the division is annotated as one frame after the actual division, then I might be able to use what is already in ACDC? If this is the case, I think it might be the easiest way to go since it is straightforward on my side and requires minimal change on ACDC. 

Cheers,
Mingwei

**Reply by ElpadoCan** [reply]
Hi Min,

You are right, I didn't think about it. If you create the tabular data that way then you should be able to use it right away. However, first I need to add the option to create this single frame in which the cells are related but not divided yet.

If you create such a table and send it to me I can then adapt the utility by adding this option. 

Cheers,
Francesco

**Comment by mm809** [comment]
Hi Francesco,

Please find the modified table attached. I put the parent ID at one frame
after the division. For this to work, I removed all daughters that contain
only one tracked frame. So there are a few cases where there is only one
daughter after the division. I hope this is not a problem.

Thanks a lot,
Mingwei
